<a href="https://colab.research.google.com/github/Alyssasorensen/datasci_2_manipulation/blob/main/datasci_2_manipulation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Cleaning

For this week's homework, I will be data cleaning with Pandas, performing data transformation, and displaying alternative libraries.

### Loading Pandas Packages

In [1]:
import pandas as pd

### Loading Data

In [2]:
healthcare_data_cleaning = pd.read_csv('https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv')

In [3]:
healthcare_data_cleaning

Patient Age  Gender  City of Residence State of Residence  \
0               45   Other         West Brian           Kentucky   
1               25  Female   East Jocelynfurt           Colorado   
2               51   Other  South Lindseyland               Ohio   
3               18  Female         Taylorfort       Pennsylvania   
4               48  Female          Birdmouth            Montana   
...            ...     ...                ...                ...   
104995          74   Other       West Anthony           Oklahoma   
104996          37    Male          Mckeetown               Utah   
104997          51  Female   Lake Lisachester           Nebraska   
104998     missing  Female       Heidiborough            missing   
104999          45  Female      South Jeffrey      Massachusetts   

       Has Insurance Visited Last Month Payment Method Preferred Doctor  \
0                Yes                Yes      Insurance     Dr. Williams   
1                 No                Yes           Card      Dr. Johnson   
2                Yes                 No           Cash     Dr. Williams   
3                Yes            missing           Cash     Dr. Williams   
4                 No                 No      Insurance        Dr. Brown   
...              ...                ...            ...              ...   
104995           Yes                 No           Card     Dr. Williams   
104996           Yes                 No           Cash        Dr. Jones   
104997           Yes                Yes      Insurance     Dr. Williams   
104998           Yes                 No           Cash        Dr. Smith   
104999            No                Yes           Cash        Dr. Brown   

       Disease Diagnosed Medication Prescribed Type of Appointment  \
0                    Flu                 Med_E             General   
1                    Flu                 Med_B          Specialist   
2               Covid-19                 Med_D             General   
3                    Flu                 Med_A             General   
4                    NaN                 Med_C          Specialist   
...                  ...                   ...                 ...   
104995          Covid-19                 Med_E          Specialist   
104996          Covid-19                 Med_C             General   
104997           Allergy                 Med_D           Follow-Up   
104998          Covid-19                 Med_A           Follow-Up   
104999           Allergy                 Med_D             General   

       Average Heart Rate Average BP Height (in cm) Weight (in kg)  \
0                      66        111            183             70   
1                      59    missing            174             58   
2                      79        119            161             56   
3                      99        115            171             52   
4                      93         89            186             87   
...                   ...        ...            ...            ...   
104995                 65        112            169             61   
104996            missing        111            165             67   
104997                 89        118            179             71   
104998                 74        123            167             75   
104999                 77        140            167             62   

           Payment Due ($) Last Visit (days ago) Visit Duration (mins)  \
0        65.45721578126224                   193                    92   
1       430.68365678679174                   195               missing   
2        315.0709305262176               missing                    37   
3        320.2998987723972                   175                   108   
4        264.2147372473319                   188                   113   
...                    ...                   ...                   ...   
104995  309.16444606680665                   183                    10   
104996         

In [4]:
data_types= healthcare_data_cleaning.dtypes
data_types

Patient Age              object
Gender                   object
City of Residence        object
State of Residence       object
Has Insurance            object
Visited Last Month       object
Payment Method           object
Preferred Doctor         object
Disease Diagnosed        object
Medication Prescribed    object
Type of Appointment      object
Average Heart Rate       object
Average BP               object
Height (in cm)           object
Weight (in kg)           object
Payment Due ($)          object
Last Visit (days ago)    object
Visit Duration (mins)    object
Number of Tests          object
Prescription Cost ($)    object
dtype: object

### Removing Duplicate Rows

In [5]:
healthcare_data_cleaning.drop_duplicates(inplace=True)
healthcare_data_cleaning

Patient Age  Gender  City of Residence State of Residence  \
0               45   Other         West Brian           Kentucky   
1               25  Female   East Jocelynfurt           Colorado   
2               51   Other  South Lindseyland               Ohio   
3               18  Female         Taylorfort       Pennsylvania   
4               48  Female          Birdmouth            Montana   
...            ...     ...                ...                ...   
104994          53  Female      Michelleburgh       South Dakota   
104995          74   Other       West Anthony           Oklahoma   
104996          37    Male          Mckeetown               Utah   
104997          51  Female   Lake Lisachester           Nebraska   
104998     missing  Female       Heidiborough            missing   

       Has Insurance Visited Last Month Payment Method Preferred Doctor  \
0                Yes                Yes      Insurance     Dr. Williams   
1                 No                Yes           Card      Dr. Johnson   
2                Yes                 No           Cash     Dr. Williams   
3                Yes            missing           Cash     Dr. Williams   
4                 No                 No      Insurance        Dr. Brown   
...              ...                ...            ...              ...   
104994            No                 No           Card     Dr. Williams   
104995           Yes                 No           Card     Dr. Williams   
104996           Yes                 No           Cash        Dr. Jones   
104997           Yes                Yes      Insurance     Dr. Williams   
104998           Yes                 No           Cash        Dr. Smith   

       Disease Diagnosed Medication Prescribed Type of Appointment  \
0                    Flu                 Med_E             General   
1                    Flu                 Med_B          Specialist   
2               Covid-19                 Med_D             General   
3                    Flu                 Med_A             General   
4                    NaN                 Med_C          Specialist   
...                  ...                   ...                 ...   
104994               Flu                 Med_B           Emergency   
104995          Covid-19                 Med_E          Specialist   
104996          Covid-19                 Med_C             General   
104997           Allergy                 Med_D           Follow-Up   
104998          Covid-19                 Med_A           Follow-Up   

       Average Heart Rate Average BP Height (in cm) Weight (in kg)  \
0                      66        111            183             70   
1                      59    missing            174             58   
2                      79        119            161             56   
3                      99        115            171             52   
4                      93         89            186             87   
...                   ...        ...            ...            ...   
104994                 68        109            172             75   
104995                 65        112            169             61   
104996            missing        111            165             67   
104997                 89        118            179             71   
104998                 74        123            167             75   

           Payment Due ($) Last Visit (days ago) Visit Duration (mins)  \
0        65.45721578126224                   193                    92   
1       430.68365678679174                   195               missing   
2        315.0709305262176               missing                    37   
3        320.2998987723972                   175                   108   
4        264.2147372473319                   188                   113   
...                    ...                   ...                   ...   
104994   488.0316041186948                   167                    25   
104995  309.164

### Removing White Space and Special Characters

In [6]:
import re  # Import the 're' module

# Function to remove white space and special characters from a value
def clean_column_names(df):
    # Define a helper function to clean column names
    def clean_name(name):
        cleaned_name = re.sub(r'[^a-zA-Z0-9]', '', name)
        return cleaned_name.lower()

    # Rename columns using the helper function
    # This is using a list comprehension - e.g., we have a list to the right of the equals sign,
    # and inside the list, we are applying our function for every col (or X) that exists in df.columns
    healthcare_data_cleaning.columns = [clean_name(col) for col in healthcare_data_cleaning.columns]  # Use 'df' instead of 'healthcare_data_cleaning'
    return healthcare_data_cleaning

# Apply the clean_column_names function to all columns
data_types = clean_column_names(data_types)

print(data_types)


       patientage  gender    cityofresidence stateofresidence hasinsurance  \
0              45   Other         West Brian         Kentucky          Yes   
1              25  Female   East Jocelynfurt         Colorado           No   
2              51   Other  South Lindseyland             Ohio          Yes   
3              18  Female         Taylorfort     Pennsylvania          Yes   
4              48  Female          Birdmouth          Montana           No   
...           ...     ...                ...              ...          ...   
104994         53  Female      Michelleburgh     South Dakota           No   
104995         74   Other       West Anthony         Oklahoma          Yes   
104996         37    Male          Mckeetown             Utah          Yes   
104997         51  Female   Lake Lisachester         Nebraska          Yes   
104998    missing  Female       Heidiborough          missing          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasedi

### Adding the Decimals Back to the "Payment Due" Column and "Prescription Cost" Column

In [7]:
import re
import numpy as np

In [8]:
healthcare_data_cleaning['paymentdue'] = pd.to_numeric(healthcare_data_cleaning['paymentdue'], errors='coerce')
healthcare_data_cleaning['prescriptioncost'] = pd.to_numeric(healthcare_data_cleaning['prescriptioncost'], errors='coerce')

In [9]:
healthcare_data_cleaning['patientage'] = pd.to_numeric(healthcare_data_cleaning['patientage'], errors='coerce')
healthcare_data_cleaning['averageheartrate'] = pd.to_numeric(healthcare_data_cleaning['averageheartrate'], errors='coerce')
healthcare_data_cleaning['averagebp'] = pd.to_numeric(healthcare_data_cleaning['averagebp'], errors='coerce')
healthcare_data_cleaning['heightincm'] = pd.to_numeric(healthcare_data_cleaning['heightincm'], errors='coerce')
healthcare_data_cleaning['weightinkg'] = pd.to_numeric(healthcare_data_cleaning['weightinkg'], errors='coerce')

In [10]:
healthcare_data_cleaning['lastvisitdaysago'] = pd.to_numeric(healthcare_data_cleaning['lastvisitdaysago'], errors='coerce')
healthcare_data_cleaning['visitdurationmins'] = pd.to_numeric(healthcare_data_cleaning['visitdurationmins'], errors='coerce')
healthcare_data_cleaning['numberoftests'] = pd.to_numeric(healthcare_data_cleaning['numberoftests'], errors='coerce')

### Handling Missing Values

In [11]:
print(data_types['patientage'].value_counts())

patientage
18.0     4047
47.0     2555
45.0     2543
44.0     2529
42.0     2518
         ... 
100.0      10
96.0       10
97.0        8
99.0        7
98.0        5
Name: count, Length: 83, dtype: int64


In [12]:
print(data_types['gender'].value_counts())

gender
Male       31728
Female     31674
Other      31598
missing     5000
Name: count, dtype: int64


In [13]:
print(data_types['cityofresidence'].value_counts())

cityofresidence
missing              5000
Lake Michael          100
West Michael           82
East Michael           77
Port Michael           74
                     ... 
Deleontown              1
Lake Hollychester       1
Port Edwardhaven        1
West Pamelaton          1
Mckeetown               1
Name: count, Length: 37133, dtype: int64


In [14]:
print(data_types['stateofresidence'].value_counts())

stateofresidence
missing           5000
New Mexico        1969
Nevada            1969
Idaho             1965
Massachusetts     1959
California        1952
Ohio              1950
Connecticut       1950
Kentucky          1946
North Carolina    1944
New Jersey        1932
Alaska            1927
Wisconsin         1926
Nebraska          1922
Rhode Island      1918
Minnesota         1918
Wyoming           1917
West Virginia     1916
Arizona           1915
Tennessee         1914
South Carolina    1912
North Dakota      1911
Virginia          1911
Utah              1910
Kansas            1909
Pennsylvania      1908
Missouri          1906
Iowa              1906
Montana           1904
Mississippi       1897
Florida           1893
Louisiana         1893
Illinois          1891
New York          1891
Oregon            1888
Indiana           1886
Vermont           1885
Texas             1876
Oklahoma          1876
Alabama           1875
Michigan          1863
Georgia           1862
Colorado         

In [15]:
print(data_types['hasinsurance'].value_counts())

hasinsurance
Yes        47597
No         47403
missing     5000
Name: count, dtype: int64


In [16]:
print(data_types['visitedlastmonth'].value_counts())

visitedlastmonth
No         47572
Yes        47428
missing     5000
Name: count, dtype: int64


In [17]:
print(data_types['paymentmethod'].value_counts())

paymentmethod
Card         31744
Insurance    31662
Cash         31594
missing       5000
Name: count, dtype: int64


In [18]:
print(data_types['preferreddoctor'].value_counts())

preferreddoctor
Dr. Jones       19153
Dr. Williams    19078
Dr. Johnson     19005
Dr. Smith       18914
Dr. Brown       18850
missing          5000
Name: count, dtype: int64


In [19]:
print(data_types['diseasediagnosed'].value_counts())

diseasediagnosed
Flu         19123
Allergy     19063
Cold        18962
Covid-19    18831
missing      5000
Name: count, dtype: int64


In [20]:
print(data_types['medicationprescribed'].value_counts())

medicationprescribed
Med_C      19149
Med_D      19040
Med_B      19003
Med_A      18957
Med_E      18851
missing     5000
Name: count, dtype: int64


In [21]:
print(data_types['typeofappointment'].value_counts())

typeofappointment
Specialist    23841
Follow-Up     23832
Emergency     23737
General       23590
missing        5000
Name: count, dtype: int64


In [22]:
print(data_types['averageheartrate'].value_counts())

averageheartrate
80.0     3779
81.0     3769
78.0     3764
77.0     3711
82.0     3694
         ... 
114.0      11
116.0       6
120.0       4
117.0       3
119.0       1
Name: count, Length: 70, dtype: int64


In [23]:
print(data_types['averagebp'].value_counts())

averagebp
80.0     2560
110.0    2509
109.0    2509
112.0    2491
111.0    2489
         ... 
82.0      489
137.0     465
81.0      425
138.0     415
139.0     374
Name: count, Length: 61, dtype: int64


In [24]:
print(data_types['heightincm'].value_counts())

heightincm
174.0    3834
175.0    3767
176.0    3731
177.0    3727
173.0    3702
         ... 
142.0      19
207.0      17
141.0      15
208.0      12
209.0      11
Name: count, Length: 71, dtype: int64


In [25]:
print(data_types['weightinkg'].value_counts())

weightinkg
71.0     2559
40.0     2544
67.0     2532
72.0     2526
69.0     2481
         ... 
134.0       1
130.0       1
132.0       1
128.0       1
135.0       1
Name: count, Length: 93, dtype: int64


In [26]:
print(data_types['paymentdue'].value_counts())

paymentdue
65.457216     1
299.988500    1
150.184654    1
297.964090    1
29.961924     1
             ..
471.830263    1
456.186160    1
273.771756    1
315.289042    1
5.341231      1
Name: count, Length: 95000, dtype: int64


In [27]:
print(data_types['lastvisitdaysago'].value_counts())

lastvisitdaysago
183.0    3992
182.0    3985
184.0    3927
180.0    3789
185.0    3777
         ... 
223.0       2
140.0       1
220.0       1
221.0       1
222.0       1
Name: count, Length: 81, dtype: int64


In [28]:
print(data_types['visitdurationmins'].value_counts())

visitdurationmins
49.0     931
69.0     924
68.0     917
27.0     914
36.0     906
        ... 
98.0     816
109.0    809
87.0     808
67.0     804
107.0    802
Name: count, Length: 111, dtype: int64


In [29]:
print(data_types['numberoftests'].value_counts())

numberoftests
4.0     23795
3.0     20400
5.0     19094
2.0     11541
6.0     10566
7.0      3993
1.0      3925
8.0       979
0.0       537
9.0       160
10.0       10
Name: count, dtype: int64


In [30]:
print(data_types['prescriptioncost'].value_counts())

prescriptioncost
15.717517     1
65.992965     1
134.851044    1
67.139600     1
4.058180      1
             ..
86.083105     1
50.488526     1
78.533369     1
124.019842    1
124.408341    1
Name: count, Length: 95000, dtype: int64


In [31]:
missing_values = healthcare_data_cleaning.isnull().sum()
missing_values

patientage               5000
gender                      0
cityofresidence             0
stateofresidence            0
hasinsurance                0
visitedlastmonth            0
paymentmethod               0
preferreddoctor             0
diseasediagnosed        19021
medicationprescribed        0
typeofappointment           0
averageheartrate         5000
averagebp                5000
heightincm               5000
weightinkg               5000
paymentdue               5000
lastvisitdaysago         5000
visitdurationmins        5000
numberoftests            5000
prescriptioncost         5000
dtype: int64

In [32]:
healthcare_data_cleaning = healthcare_data_cleaning.replace('missing', 'Unknown')
healthcare_data_cleaning

patientage  gender    cityofresidence stateofresidence hasinsurance  \
0             45.0   Other         West Brian         Kentucky          Yes   
1             25.0  Female   East Jocelynfurt         Colorado           No   
2             51.0   Other  South Lindseyland             Ohio          Yes   
3             18.0  Female         Taylorfort     Pennsylvania          Yes   
4             48.0  Female          Birdmouth          Montana           No   
...            ...     ...                ...              ...          ...   
104994        53.0  Female      Michelleburgh     South Dakota           No   
104995        74.0   Other       West Anthony         Oklahoma          Yes   
104996        37.0    Male          Mckeetown             Utah          Yes   
104997        51.0  Female   Lake Lisachester         Nebraska          Yes   
104998         NaN  Female       Heidiborough          Unknown          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasediagnosed  \
0                   Yes     Insurance    Dr. Williams              Flu   
1                   Yes          Card     Dr. Johnson              Flu   
2                    No          Cash    Dr. Williams         Covid-19   
3               Unknown          Cash    Dr. Williams              Flu   
4                    No     Insurance       Dr. Brown              NaN   
...                 ...           ...             ...              ...   
104994               No          Card    Dr. Williams              Flu   
104995               No          Card    Dr. Williams         Covid-19   
104996               No          Cash       Dr. Jones         Covid-19   
104997              Yes     Insurance    Dr. Williams          Allergy   
104998               No          Cash       Dr. Smith         Covid-19   

       medicationprescribed typeofappointment  averageheartrate  averagebp  \
0                     Med_E           General              66.0      111.0   
1                     Med_B        Specialist              59.0        NaN   
2                     Med_D           General              79.0      119.0   
3                     Med_A           General              99.0      115.0   
4                     Med_C        Specialist              93.0       89.0   
...                     ...               ...               ...        ...   
104994                Med_B         Emergency              68.0      109.0   
104995                Med_E        Specialist              65.0      112.0   
104996                Med_C           General               NaN      111.0   
104997                Med_D         Follow-Up              89.0      118.0   
104998                Med_A         Follow-Up              74.0      123.0   

        heightincm  weightinkg  paymentdue  lastvisitdaysago  \
0            183.0        70.0   65.457216             193.0   
1            174.0        58.0  430.683657             195.0   
2            161.0        56.0  315.070931               NaN   
3            171.0        52.0  320.299899             175.0   
4            186.0        87.0  264.214737             188.0   
...            ...         ...         ...               ...   
104994       172.0        75.0  488.031604             167.0   
104995       169.0        61.0  309.164446             183.0   
104996       165.0        67.0         NaN             191.0   
104997       179.0        71.0  320.830418             172.0   
104998       167.0        75.0    5.341231             180.0   

        visitdurationmins  numberoftests  prescriptioncost  
0                    92.0            5.0         15.717517  
1                     NaN            4.0         80.786473  
2                    37.0            2.0         64.392139  
3                   108.0            2.0          8.872859  
4                   113.0            3.0         77.483113  
...                   ...            ...               ...  
104994               25.0            

### Cleaning Column Names

In [33]:
healthcare_data_cleaning.columns = healthcare_data_cleaning.columns.str.strip().str.lower().str.replace(' ', '_').str.replace('(', '').str.replace(')', '')
healthcare_data_cleaning

patientage  gender    cityofresidence stateofresidence hasinsurance  \
0             45.0   Other         West Brian         Kentucky          Yes   
1             25.0  Female   East Jocelynfurt         Colorado           No   
2             51.0   Other  South Lindseyland             Ohio          Yes   
3             18.0  Female         Taylorfort     Pennsylvania          Yes   
4             48.0  Female          Birdmouth          Montana           No   
...            ...     ...                ...              ...          ...   
104994        53.0  Female      Michelleburgh     South Dakota           No   
104995        74.0   Other       West Anthony         Oklahoma          Yes   
104996        37.0    Male          Mckeetown             Utah          Yes   
104997        51.0  Female   Lake Lisachester         Nebraska          Yes   
104998         NaN  Female       Heidiborough          Unknown          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasediagnosed  \
0                   Yes     Insurance    Dr. Williams              Flu   
1                   Yes          Card     Dr. Johnson              Flu   
2                    No          Cash    Dr. Williams         Covid-19   
3               Unknown          Cash    Dr. Williams              Flu   
4                    No     Insurance       Dr. Brown              NaN   
...                 ...           ...             ...              ...   
104994               No          Card    Dr. Williams              Flu   
104995               No          Card    Dr. Williams         Covid-19   
104996               No          Cash       Dr. Jones         Covid-19   
104997              Yes     Insurance    Dr. Williams          Allergy   
104998               No          Cash       Dr. Smith         Covid-19   

       medicationprescribed typeofappointment  averageheartrate  averagebp  \
0                     Med_E           General              66.0      111.0   
1                     Med_B        Specialist              59.0        NaN   
2                     Med_D           General              79.0      119.0   
3                     Med_A           General              99.0      115.0   
4                     Med_C        Specialist              93.0       89.0   
...                     ...               ...               ...        ...   
104994                Med_B         Emergency              68.0      109.0   
104995                Med_E        Specialist              65.0      112.0   
104996                Med_C           General               NaN      111.0   
104997                Med_D         Follow-Up              89.0      118.0   
104998                Med_A         Follow-Up              74.0      123.0   

        heightincm  weightinkg  paymentdue  lastvisitdaysago  \
0            183.0        70.0   65.457216             193.0   
1            174.0        58.0  430.683657             195.0   
2            161.0        56.0  315.070931               NaN   
3            171.0        52.0  320.299899             175.0   
4            186.0        87.0  264.214737             188.0   
...            ...         ...         ...               ...   
104994       172.0        75.0  488.031604             167.0   
104995       169.0        61.0  309.164446             183.0   
104996       165.0        67.0         NaN             191.0   
104997       179.0        71.0  320.830418             172.0   
104998       167.0        75.0    5.341231             180.0   

        visitdurationmins  numberoftests  prescriptioncost  
0                    92.0            5.0         15.717517  
1                     NaN            4.0         80.786473  
2                    37.0            2.0         64.392139  
3                   108.0            2.0          8.872859  
4                   113.0            3.0         77.483113  
...                   ...            ...               ...  
104994               25.0            

### Summary of the Changes Made

##### The initial step involved loading the packages into Google Colab. This meant that I needed to write "import pandas as pd." Following this, I had to load the data. I inserted the CSV file from GitHub and Google Colab displayed the data. After this, I removed duplicate rows from the DataFrame. This was done by inserting, "healthcare_data_cleaning.drop_duplicates(inplace=True)." The "drop_duplicates" identifies and removes duplicate rows from the DataFrame based on the columns' values by default. The statement "inplace=True" is an argument passed to the "drop_duplicates" method. When "inplace=True" is set, it means the operation wil be performed directly on the original DataFrame, which is healthcare_data_cleaning. After this, I removed white spaces and special characters. The function inserted for this was used to ensure that column names in the DataFrame do not contain non-alphanumeric characters, white spaces, and are standardized to lowercase, which can be helpful for consistency. Next, I added the decimals back to the values of payment due and prescription cost columns. Next, I handled the missing values from the DataFrame. I did this by inserting "print(data_types['patientage'].value_counts())." Following this, I cleaned some of the column names by changing how they are displayed in the DataFrame.

# Data Transformation




### Creating New Columns Based on Existing Ones

In [34]:
healthcare_data_cleaning.rename(columns = {'patient_age':'age', 'city_of_residence': 'city', 'state_of_residence': 'state', 'disease_diagnosed':'diagnosis', 'medication_prescribed': 'prescription', 'height_in_cm': 'height_cm', 'weight_in_kg':'weight_kg'}, inplace = True)
healthcare_data_cleaning

patientage  gender    cityofresidence stateofresidence hasinsurance  \
0             45.0   Other         West Brian         Kentucky          Yes   
1             25.0  Female   East Jocelynfurt         Colorado           No   
2             51.0   Other  South Lindseyland             Ohio          Yes   
3             18.0  Female         Taylorfort     Pennsylvania          Yes   
4             48.0  Female          Birdmouth          Montana           No   
...            ...     ...                ...              ...          ...   
104994        53.0  Female      Michelleburgh     South Dakota           No   
104995        74.0   Other       West Anthony         Oklahoma          Yes   
104996        37.0    Male          Mckeetown             Utah          Yes   
104997        51.0  Female   Lake Lisachester         Nebraska          Yes   
104998         NaN  Female       Heidiborough          Unknown          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasediagnosed  \
0                   Yes     Insurance    Dr. Williams              Flu   
1                   Yes          Card     Dr. Johnson              Flu   
2                    No          Cash    Dr. Williams         Covid-19   
3               Unknown          Cash    Dr. Williams              Flu   
4                    No     Insurance       Dr. Brown              NaN   
...                 ...           ...             ...              ...   
104994               No          Card    Dr. Williams              Flu   
104995               No          Card    Dr. Williams         Covid-19   
104996               No          Cash       Dr. Jones         Covid-19   
104997              Yes     Insurance    Dr. Williams          Allergy   
104998               No          Cash       Dr. Smith         Covid-19   

       medicationprescribed typeofappointment  averageheartrate  averagebp  \
0                     Med_E           General              66.0      111.0   
1                     Med_B        Specialist              59.0        NaN   
2                     Med_D           General              79.0      119.0   
3                     Med_A           General              99.0      115.0   
4                     Med_C        Specialist              93.0       89.0   
...                     ...               ...               ...        ...   
104994                Med_B         Emergency              68.0      109.0   
104995                Med_E        Specialist              65.0      112.0   
104996                Med_C           General               NaN      111.0   
104997                Med_D         Follow-Up              89.0      118.0   
104998                Med_A         Follow-Up              74.0      123.0   

        heightincm  weightinkg  paymentdue  lastvisitdaysago  \
0            183.0        70.0   65.457216             193.0   
1            174.0        58.0  430.683657             195.0   
2            161.0        56.0  315.070931               NaN   
3            171.0        52.0  320.299899             175.0   
4            186.0        87.0  264.214737             188.0   
...            ...         ...         ...               ...   
104994       172.0        75.0  488.031604             167.0   
104995       169.0        61.0  309.164446             183.0   
104996       165.0        67.0         NaN             191.0   
104997       179.0        71.0  320.830418             172.0   
104998       167.0        75.0    5.341231             180.0   

        visitdurationmins  numberoftests  prescriptioncost  
0                    92.0            5.0         15.717517  
1                     NaN            4.0         80.786473  
2                    37.0            2.0         64.392139  
3                   108.0            2.0          8.872859  
4                   113.0            3.0         77.483113  
...                   ...            ...               ...  
104994               25.0            

### Aggregating Data Using Groupby and Compute Summary Statistics

In [35]:
healthcare_data_cleaning['diseasediagnosed'].value_counts()

diseasediagnosed
Flu         19123
Allergy     19063
Cold        18962
Covid-19    18831
Unknown      5000
Name: count, dtype: int64

In [36]:
encompass_diagnosis = healthcare_data_cleaning[healthcare_data_cleaning['diseasediagnosed'] == 'Flu']
encompass_diagnosis.groupby(['diseasediagnosed']).sum()

patientage  \
diseasediagnosed               
Flu                 811046.0   

                                                             gender  \
diseasediagnosed                                                      
Flu               OtherFemaleFemaleMaleFemaleFemaleMaleOtherMale...   

                                                    cityofresidence  \
diseasediagnosed                                                      
Flu               West BrianEast JocelynfurtTaylorfortVanessabur...   

                                                   stateofresidence  \
diseasediagnosed                                                      
Flu               KentuckyColoradoPennsylvaniaConnecticutMontana...   

                                                       hasinsurance  \
diseasediagnosed                                                      
Flu               YesNoYesYesNoYesNoNoNoNoYesNoNoYesYesYesYesNoN...   

                                                   visitedlastmonth  \
diseasediagnosed                                                      
Flu               YesYesUnknownYesYesYesYesYesNoYesYesNoYesNoYes...   

                                                      paymentmethod  \
diseasediagnosed                                                      
Flu               InsuranceCardCashInsuranceCardCashUnknownCashC...   

                                                    preferreddoctor  \
diseasediagnosed                                                      
Flu               Dr. WilliamsDr. JohnsonDr. WilliamsDr. William...   

                                               medicationprescribed  \
diseasediagnosed                                                      
Flu               Med_EMed_BMed_AMed_EMed_CUnknownMed_AUnknownUn...   

                                                  typeofappointment  \
diseasediagnosed                                                      
Flu               GeneralSpecialistGeneralGeneralSpecialistGener...   

                  averageheartrate  averagebp  heightincm  weightinkg  \
diseasediagnosed                                                        
Flu                      1442699.0  1987959.0   3168630.0   1262238.0   

                    paymentdue  lastvisitdaysago  visitdurationmins  \
diseasediagnosed                                                      
Flu               4.500347e+06         3313991.0          1179560.0   

                  numberoftests  prescriptioncost  
diseasediagnosed                                   
Flu                     72505.0      1.359500e+06

In [37]:
healthcare_data_cleaning

patientage  gender    cityofresidence stateofresidence hasinsurance  \
0             45.0   Other         West Brian         Kentucky          Yes   
1             25.0  Female   East Jocelynfurt         Colorado           No   
2             51.0   Other  South Lindseyland             Ohio          Yes   
3             18.0  Female         Taylorfort     Pennsylvania          Yes   
4             48.0  Female          Birdmouth          Montana           No   
...            ...     ...                ...              ...          ...   
104994        53.0  Female      Michelleburgh     South Dakota           No   
104995        74.0   Other       West Anthony         Oklahoma          Yes   
104996        37.0    Male          Mckeetown             Utah          Yes   
104997        51.0  Female   Lake Lisachester         Nebraska          Yes   
104998         NaN  Female       Heidiborough          Unknown          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasediagnosed  \
0                   Yes     Insurance    Dr. Williams              Flu   
1                   Yes          Card     Dr. Johnson              Flu   
2                    No          Cash    Dr. Williams         Covid-19   
3               Unknown          Cash    Dr. Williams              Flu   
4                    No     Insurance       Dr. Brown              NaN   
...                 ...           ...             ...              ...   
104994               No          Card    Dr. Williams              Flu   
104995               No          Card    Dr. Williams         Covid-19   
104996               No          Cash       Dr. Jones         Covid-19   
104997              Yes     Insurance    Dr. Williams          Allergy   
104998               No          Cash       Dr. Smith         Covid-19   

       medicationprescribed typeofappointment  averageheartrate  averagebp  \
0                     Med_E           General              66.0      111.0   
1                     Med_B        Specialist              59.0        NaN   
2                     Med_D           General              79.0      119.0   
3                     Med_A           General              99.0      115.0   
4                     Med_C        Specialist              93.0       89.0   
...                     ...               ...               ...        ...   
104994                Med_B         Emergency              68.0      109.0   
104995                Med_E        Specialist              65.0      112.0   
104996                Med_C           General               NaN      111.0   
104997                Med_D         Follow-Up              89.0      118.0   
104998                Med_A         Follow-Up              74.0      123.0   

        heightincm  weightinkg  paymentdue  lastvisitdaysago  \
0            183.0        70.0   65.457216             193.0   
1            174.0        58.0  430.683657             195.0   
2            161.0        56.0  315.070931               NaN   
3            171.0        52.0  320.299899             175.0   
4            186.0        87.0  264.214737             188.0   
...            ...         ...         ...               ...   
104994       172.0        75.0  488.031604             167.0   
104995       169.0        61.0  309.164446             183.0   
104996       165.0        67.0         NaN             191.0   
104997       179.0        71.0  320.830418             172.0   
104998       167.0        75.0    5.341231             180.0   

        visitdurationmins  numberoftests  prescriptioncost  
0                    92.0            5.0         15.717517  
1                     NaN            4.0         80.786473  
2                    37.0            2.0         64.392139  
3                   108.0            2.0          8.872859  
4                   113.0            3.0         77.483113  
...                   ...            ...               ...  
104994               25.0            

In [38]:
encompass_diagnosis = healthcare_data_cleaning[healthcare_data_cleaning['diseasediagnosed'] == 'Covid-19']
encompass_diagnosis.groupby(['diseasediagnosed']).sum()

patientage  \
diseasediagnosed               
Covid-19            796229.0   

                                                             gender  \
diseasediagnosed                                                      
Covid-19          OtherUnknownMaleFemaleMaleFemaleMaleUnknownOth...   

                                                    cityofresidence  \
diseasediagnosed                                                      
Covid-19          South LindseylandNorth DonaldLake JenniferJone...   

                                                   stateofresidence  \
diseasediagnosed                                                      
Covid-19          OhioOklahomaUnknownRhode IslandOregonNew Mexic...   

                                                       hasinsurance  \
diseasediagnosed                                                      
Covid-19          YesNoNoNoYesNoYesNoNoUnknownYesNoUnknownNoYesN...   

                                                   visitedlastmonth  \
diseasediagnosed                                                      
Covid-19          NoYesNoNoYesYesYesNoYesYesNoYesNoNoNoYesYesYes...   

                                                      paymentmethod  \
diseasediagnosed                                                      
Covid-19          CashCardCashCardCardInsuranceInsuranceCardCard...   

                                                    preferreddoctor  \
diseasediagnosed                                                      
Covid-19          Dr. WilliamsDr. JonesUnknownDr. JonesDr. Jones...   

                                               medicationprescribed  \
diseasediagnosed                                                      
Covid-19          Med_DMed_BMed_AMed_EMed_BMed_CMed_DMed_BMed_EM...   

                                                  typeofappointment  \
diseasediagnosed                                                      
Covid-19          GeneralGeneralEmergencySpecialistSpecialistEme...   

                  averageheartrate  averagebp  heightincm  weightinkg  \
diseasediagnosed                                                        
Covid-19                 1423634.0  1954912.0   3133540.0   1241259.0   

                    paymentdue  lastvisitdaysago  visitdurationmins  \
diseasediagnosed                                                      
Covid-19          4.478499e+06         3272427.0          1162782.0   

                  numberoftests  prescriptioncost  
diseasediagnosed                                   
Covid-19                71489.0      1.343962e+06

In [39]:
healthcare_data_cleaning

patientage  gender    cityofresidence stateofresidence hasinsurance  \
0             45.0   Other         West Brian         Kentucky          Yes   
1             25.0  Female   East Jocelynfurt         Colorado           No   
2             51.0   Other  South Lindseyland             Ohio          Yes   
3             18.0  Female         Taylorfort     Pennsylvania          Yes   
4             48.0  Female          Birdmouth          Montana           No   
...            ...     ...                ...              ...          ...   
104994        53.0  Female      Michelleburgh     South Dakota           No   
104995        74.0   Other       West Anthony         Oklahoma          Yes   
104996        37.0    Male          Mckeetown             Utah          Yes   
104997        51.0  Female   Lake Lisachester         Nebraska          Yes   
104998         NaN  Female       Heidiborough          Unknown          Yes   

       visitedlastmonth paymentmethod preferreddoctor diseasediagnosed  \
0                   Yes     Insurance    Dr. Williams              Flu   
1                   Yes          Card     Dr. Johnson              Flu   
2                    No          Cash    Dr. Williams         Covid-19   
3               Unknown          Cash    Dr. Williams              Flu   
4                    No     Insurance       Dr. Brown              NaN   
...                 ...           ...             ...              ...   
104994               No          Card    Dr. Williams              Flu   
104995               No          Card    Dr. Williams         Covid-19   
104996               No          Cash       Dr. Jones         Covid-19   
104997              Yes     Insurance    Dr. Williams          Allergy   
104998               No          Cash       Dr. Smith         Covid-19   

       medicationprescribed typeofappointment  averageheartrate  averagebp  \
0                     Med_E           General              66.0      111.0   
1                     Med_B        Specialist              59.0        NaN   
2                     Med_D           General              79.0      119.0   
3                     Med_A           General              99.0      115.0   
4                     Med_C        Specialist              93.0       89.0   
...                     ...               ...               ...        ...   
104994                Med_B         Emergency              68.0      109.0   
104995                Med_E        Specialist              65.0      112.0   
104996                Med_C           General               NaN      111.0   
104997                Med_D         Follow-Up              89.0      118.0   
104998                Med_A         Follow-Up              74.0      123.0   

        heightincm  weightinkg  paymentdue  lastvisitdaysago  \
0            183.0        70.0   65.457216             193.0   
1            174.0        58.0  430.683657             195.0   
2            161.0        56.0  315.070931               NaN   
3            171.0        52.0  320.299899             175.0   
4            186.0        87.0  264.214737             188.0   
...            ...         ...         ...               ...   
104994       172.0        75.0  488.031604             167.0   
104995       169.0        61.0  309.164446             183.0   
104996       165.0        67.0         NaN             191.0   
104997       179.0        71.0  320.830418             172.0   
104998       167.0        75.0    5.341231             180.0   

        visitdurationmins  numberoftests  prescriptioncost  
0                    92.0            5.0         15.717517  
1                     NaN            4.0         80.786473  
2                    37.0            2.0         64.392139  
3                   108.0            2.0          8.872859  
4                   113.0            3.0         77.483113  
...                   ...            ...               ...  
104994               25.0            

In [40]:
healthcare_data_cleaning.columns

Index(['patientage', 'gender', 'cityofresidence', 'stateofresidence',
       'hasinsurance', 'visitedlastmonth', 'paymentmethod', 'preferreddoctor',
       'diseasediagnosed', 'medicationprescribed', 'typeofappointment',
       'averageheartrate', 'averagebp', 'heightincm', 'weightinkg',
       'paymentdue', 'lastvisitdaysago', 'visitdurationmins', 'numberoftests',
       'prescriptioncost'],
      dtype='object')

In [41]:
pivot_table = healthcare_data_cleaning.pivot_table(index='patientage', columns='gender', values='numberoftests', aggfunc='mean')
pivot_table

gender        Female      Male     Other   Unknown
patientage                                        
18.0        4.005710  4.031771  3.976311  4.031414
19.0        3.994872  3.969880  4.072917  3.884615
20.0        4.044118  3.943590  3.905941  4.000000
21.0        4.004098  4.152466  4.008368  4.268293
22.0        4.042636  3.886878  3.882562  3.745455
...              ...       ...       ...       ...
96.0        3.750000  3.000000  4.750000       NaN
97.0        4.666667  4.000000  4.000000  3.000000
98.0        4.500000  2.000000       NaN  7.000000
99.0        5.000000  2.500000  5.000000       NaN
100.0            NaN  5.166667  5.000000       NaN

[83 rows x 4 columns]

In [42]:
pivot_table = healthcare_data_cleaning.pivot_table(index='gender', columns='hasinsurance', values='paymentdue', aggfunc='mean')
pivot_table

hasinsurance          No     Unknown         Yes
gender                                          
Female        248.505546  249.640663  249.048480
Male          251.145420  246.216726  249.736330
Other         248.469690  252.076968  250.898797
Unknown       249.741155  251.416699  255.595722

In [43]:
pivot_table = healthcare_data_cleaning.pivot_table(index='diseasediagnosed', columns='typeofappointment', values='prescriptioncost', aggfunc='mean')
pivot_table

typeofappointment  Emergency  Follow-Up    General  Specialist    Unknown
diseasediagnosed                                                         
Allergy            74.661355  75.737521  74.566674   74.845884  73.246786
Cold               74.479360  75.168171  74.869570   75.225813  75.269402
Covid-19           74.155488  75.477123  74.875238   75.255247  74.074811
Flu                75.498899  74.150736  74.609996   75.972985  73.410834
Unknown            72.860005  75.094282  76.008557   73.818379  72.771497

# Introduction to Alternative Libraries

### Loading Polars Package

In [44]:
import polars as pl

### Loading Data for Polars

In [45]:
healthcare_data_cleaning = pl.read_csv('https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv')

In [46]:
healthcare_data_cleaning

shape: (105_000, 20)
┌───────────┬────────┬────────────┬────────────┬───┬────────────┬────────────┬────────────┬────────────┐
│ Patient   ┆ Gender ┆ City of    ┆ State of   ┆ … ┆ Last Visit ┆ Visit      ┆ Number of  ┆ Prescripti │
│ Age       ┆ ---    ┆ Residence  ┆ Residence  ┆   ┆ (days ago) ┆ Duration   ┆ Tests      ┆ on Cost    │
│ ---       ┆ str    ┆ ---        ┆ ---        ┆   ┆ ---        ┆ (mins)     ┆ ---        ┆ ($)        │
│ str       ┆        ┆ str        ┆ str        ┆   ┆ str        ┆ ---        ┆ str        ┆ ---        │
│           ┆        ┆            ┆            ┆   ┆            ┆ str        ┆            ┆ str        │
╞═══════════╪════════╪════════════╪════════════╪═══╪════════════╪════════════╪════════════╪════════════╡
│ 45        ┆ Other  ┆ West Brian ┆ Kentucky   ┆ … ┆ 193        ┆ 92         ┆ 5          ┆ 15.7175173 │
│           ┆        ┆            ┆            ┆   ┆            ┆            ┆            ┆ 5532889    │
│ 25        ┆ Female ┆ East Jocel ┆ Colorado   ┆ … ┆ 195        ┆ missing    ┆ 4          ┆ 80.7864728 │
│           ┆        ┆ ynfurt     ┆            ┆   ┆            ┆            ┆            ┆ 4463952    │
│ 51        ┆ Other  ┆ South Lind ┆ Ohio       ┆ … ┆ missing    ┆ 37         ┆ 2          ┆ 64.3921393 │
│           ┆        ┆ seyland    ┆            ┆   ┆            ┆            ┆            ┆ 009105     │
│ 18        ┆ Female ┆ Taylorfort ┆ Pennsylvan ┆ … ┆ 175        ┆ 108        ┆ 2          ┆ 8.87285921 │
│           ┆        ┆            ┆ ia         ┆   ┆            ┆            ┆            ┆ 5315316    │
│ …         ┆ …      ┆ …          ┆ …          ┆ … ┆ …          ┆ …          ┆ …          ┆ …          │
│ 37        ┆ Male   ┆ Mckeetown  ┆ Utah       ┆ … ┆ 191        ┆ 47         ┆ 2          ┆ 113.867640 │
│           ┆        ┆            ┆            ┆   ┆            ┆            ┆            ┆ 25370326   │
│ 51        ┆ Female ┆ Lake Lisac ┆ Nebraska   ┆ … ┆ 172        ┆ 39         ┆ 3          ┆ 35.6224202 │
│           ┆        ┆ hester     ┆            ┆   ┆            ┆            ┆            ┆ 32475356   │
│ missing   ┆ Female ┆ Heidiborou ┆ missing    ┆ … ┆ 180        ┆ 79         ┆ missing    ┆ 124.408340 │
│           ┆        ┆ gh         ┆            ┆   ┆            ┆            ┆            ┆ 9893742    │
│ 45        ┆ Female ┆ South      ┆ Massachuse ┆ … ┆ 167        ┆ 81         ┆ missing    ┆ 87.9106014 │
│           ┆        ┆ Jeffrey    ┆ tts        ┆   ┆            ┆            ┆            ┆ 9353706    │
└───────────┴────────┴────────────┴────────────┴───┴────────────┴────────────┴────────────┴────────────┘

In [64]:
import polars as pl

healthcare_data_cleaning = pl.read_csv('https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv')

print("Number of rows:", healthcare_data_cleaning.shape[0])
print("Number of columns:", healthcare_data_cleaning.shape[1])

Number of rows: 105000
Number of columns: 20


In [65]:
import time

In [66]:
dataset_path = 'healthcare_data_cleaning.csv'

In [71]:
start_time = time.time()
polars_healthcare_data_cleaning = pd.read_csv('https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv')
polars_filtered = polars_healthcare_data_cleaning[polars_healthcare_data_cleaning['Gender'] == 'Male']
polars_average_age = polars_filtered['Average BP'].sum()
polars_time = time.time() - start_time

### Loading Modin Package

In [47]:
pip install "modin[all]"

In [117]:
pip install --upgrade pandas

  Using cached pandas-2.1.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.7 MB)
  Attempting uninstall: pandas
    Found existing installation: pandas 2.0.3
    Uninstalling pandas-2.0.3:
      Successfully uninstalled pandas-2.0.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires pandas==1.5.3, but you have pandas 2.1.0 which is incompatible.
modin 0.23.1 requires pandas<2.1,>=2, but you have pandas 2.1.0 which is incompatible.


In [118]:
pip install --upgrade modin

  Using cached pandas-2.0.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)
  Attempting uninstall: pandas
    Found existing installation: pandas 2.1.0
    Uninstalling pandas-2.1.0:
      Successfully uninstalled pandas-2.1.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-colab 1.0.0 requires pandas==1.5.3, but you have pandas 2.0.3 which is incompatible.


In [119]:
pip install botocore

In [120]:
pip install -U modin

In [121]:
# Post-install automatically kill and restart Colab environment
import os

In [ ]:
os.kill(os.getpid(), 9)

In [48]:
# import pandas as pd
import modin.pandas as pd

In [49]:
import modin.pandas as mpd

In [50]:
import modin.pandas as mpd
import requests

url = 'https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv'
response = requests.get(url)

# Save the file locally
with open('healthcare_data.csv', 'wb') as file:
    file.write(response.content)

# Read the downloaded file
healthcare_data_cleaning = mpd.read_csv('healthcare_data.csv')


To remove this warning, run the following python code before doing dataframe operations:

    import ray
    ray.init()

2023-09-09 22:32:27,708	INFO worker.py:1621 -- Started a local Ray instance.


In [51]:
# Create a Modin DataFrame from a CSV file containing healthcare data
# For example, we have patient records with columns like 'Patient Age', 'Gender', 'City of Residence', 'Disease Diagnosed'
healthcare_data_cleaning = mpd.read_csv('healthcare_data.csv')

In [52]:
# Display the first few rows of the Modin DataFrame
print(healthcare_data_cleaning.head())

# Filter the data to select patients with a specific condition
disease_diagnosed_patients = healthcare_data_cleaning[healthcare_data_cleaning['Disease Diagnosed'] == 'Flu']

  Patient Age  Gender  City of Residence State of Residence Has Insurance  \
0          45   Other         West Brian           Kentucky           Yes   
1          25  Female   East Jocelynfurt           Colorado            No   
2          51   Other  South Lindseyland               Ohio           Yes   
3          18  Female         Taylorfort       Pennsylvania           Yes   
4          48  Female          Birdmouth            Montana            No   

  Visited Last Month Payment Method Preferred Doctor Disease Diagnosed  \
0                Yes      Insurance     Dr. Williams               Flu   
1                Yes           Card      Dr. Johnson               Flu   
2                 No           Cash     Dr. Williams          Covid-19   
3            missing           Cash     Dr. Williams               Flu   
4                 No      Insurance        Dr. Brown               NaN   

  Medication Prescribed Type of Appointment Average Heart Rate Average BP  \
0              

In [53]:
unique_values = disease_diagnosed_patients['Patient Age'].unique()
print(unique_values)

['45' '25' '18' '52' '20' '54' '39' '62' '72' '44' '58' '36' '42' '49'
 '40' '32' '37' '46' '34' '50' '22' '78' '66' '48' '19' '64' 'missing'
 '43' '38' '47' '55' '88' '41' '35' '59' '56' '27' '53' '31' '24' '67'
 '51' '33' '77' '68' '60' '61' '71' '30' '57' '28' '26' '79' '74' '76'
 '69' '90' '63' '75' '65' '21' '80' '70' '29' '23' '81' '73' '82' '85'
 '87' '94' '83' '92' '97' '86' '84' '89' '91' '96' '98' '99' '93' '95']


In [54]:
disease_diagnosed_patients['Patient Age'] = pd.to_numeric(disease_diagnosed_patients['Patient Age'], errors='coerce')

In [55]:
# Average age of patients with the flu
average_age = disease_diagnosed_patients['Patient Age'].mean()
print(f"Average age of patients with the flu: {average_age:.2f} years")

Average age of patients with the flu: 44.71 years


In [56]:
import modin.pandas as mpd
import pandas as pd
import time

In [72]:
# Define the dataset path
dataset_path = 'healthcare_data_cleaning.csv'

### Loading Times

In [79]:
start_time = time.time()
modin_healthcare_data_cleaning = pd.read_csv('https://raw.githubusercontent.com/Alyssasorensen/datasci_2_manipulation/main/datasets/raw.githubusercontent.com_hantswilliams_HHA_507_2023_main_WK2_data_healthcare_data_cleaning.csv')
modin_filtered = modin_healthcare_data_cleaning[modin_healthcare_data_cleaning['Gender'] == 'Male']
modin_average_age = modin_filtered['Average BP'].sum()
modin_time = time.time() - start_time

In [80]:
print(f"Modin Execution Time: {modin_time:.4f} seconds")
print(f"Polars Execution Time: {polars_time:.4f} seconds")

Modin Execution Time: 0.5426 seconds
Polars Execution Time: 0.4323 seconds


### Loading Times Observation
###### Modin took longer than Polars to execute the code. Modin took 0.5426 seconds and Polars took 0.4323 seconds. Modin took approximately 0.1103 seconds longer to execute the code.